src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBuildWrapperTest.java

```diff
@@ -0,0 +1,97 @@
+        startMockSSHServer();
+
+        List<String> credentialIds = new ArrayList<String>();
+        credentialIds.add(CREDENTIAL_ID);
+
+        SSHUserPrivateKey key = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, credentialIds.get(0), "cloudbees",
+                new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "cloudbees", "test");
+        SystemCredentialsProvider.getInstance().getCredentials().add(key);
+        SystemCredentialsProvider.getInstance().save();
+
+        FreeStyleProject job = r.createFreeStyleProject();
+
+        SSHAgentBuildWrapper sshAgent = new SSHAgentBuildWrapper(credentialIds, false);
+        job.getBuildWrappersList().add(sshAgent);
+
+        Shell shell = new Shell("ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST);
+        job.getBuildersList().add(shell);
+
+        r.assertBuildStatusSuccess(job.scheduleBuild2(0));
+
+        stopMockSSHServer();
+    }
+
+    @Test
+    public void sshAgentUnavailable() throws Exception {
+        startMockSSHServer();
+
+        List<String> credentialIds = new ArrayList<String>();
+        credentialIds.add(CREDENTIAL_ID);
+
+        SSHUserPrivateKey key = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, credentialIds.get(0), "cloudbees",
+                new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "cloudbees", "test");
+        SystemCredentialsProvider.getInstance().getCredentials().add(key);
+        SystemCredentialsProvider.getInstance().save();
+
+        FreeStyleProject job = r.createFreeStyleProject();
+
+        Shell shell = new Shell("ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST);
+        job.getBuildersList().add(shell);
+
+        r.assertLogContains("Permission denied (publickey).", r.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get()));
+
+        stopMockSSHServer();
+    }
+
+    @Test
+    public void sshAgentWithInvalidCredentials() throws Exception {
+        startMockSSHServer();
+
+        List<String> credentialIds = new ArrayList<String>();
+        credentialIds.add(CREDENTIAL_ID);
+
+        SSHUserPrivateKey key = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, credentialIds.get(0), "cloudbees",
+                new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "BAD-passphrase-cloudbees", "test");
+        SystemCredentialsProvider.getInstance().getCredentials().add(key);
+        SystemCredentialsProvider.getInstance().save();
+
+        FreeStyleProject job = r.createFreeStyleProject();
+
+        Shell shell = new Shell("ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST);
+        job.getBuildersList().add(shell);
+
+        r.assertLogContains("Permission denied (publickey).", r.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get()));
```

Comment:
This seems to be the same as `sshAgentUnavailable`: you are not adding the wrapper, so of course the `ssh` command fails.